### PR TITLE
=doc Remove outdated note in Streaming File IO

### LINKED ISSUE #19904: Outdated note in Streaming File IO
It has a note that talks about "we're still on 2.3.x" etc.

http://doc.akka.io/docs/akka/2.4.2/scala/stream/stream-io.html#Streaming_File_IO

Remove / adjust the note.

### PR DESCRIPTION
**Pull request by naferx**
Fixes #19904

### CODE REVIEW

File: akka-docs/rst/java/stream/stream-io.rst [hunk 1]
```diff
@@ -100,7 +100,7 @@ Akka Streams provide simple Sources and Sinks that can work with :class:`ByteStr
 on files.
 
 .. note::
-  Since the current version of Akka (``2.3.x``) needs to support JDK6, the currently provided File IO implementations
+  Since the version of Akka (``2.3.x``) needs to support JDK6, the currently provided File IO implementations
   are not able to utilise Asynchronous File IO operations, as these were introduced in JDK7 (and newer).
   Once Akka is free to require JDK8 (from ``2.4.x``) these implementations will be updated to make use of the
```
Comment (johanandren) [inline akka-docs/rst/java/stream/stream-io.rst]: The entire note needs updating, as we are now on 2.4.x and JDK8. We just haven't had time to move to NIO. Maybe we should just remove it.

Comment (ktoso) [inline akka-docs/rst/java/stream/stream-io.rst]: +1 to removing the entire note

**Comment by naferx** [comment]
@johanandren @ktoso Updated the PR according to your suggestions!.

**Comment by ktoso** [comment]
Thanks a lot @naferx!
LGTM

**Comment by ktoso** [comment]
Resolves https://github.com/akka/akka/issues/19942
Resolves #19904

**Comment by julianvmodesto** [comment]
Does the streaming FileIO implementation use NIO's AsynchronousFileChannel?

**Comment by ktoso** [comment]
No it doesn't and I don't think you'd notice as we dedicate a preconfigured threadpool for the blocking ops
In my benchmarks only small performance gains could be won with the Async channel, though I'd definitely welcome a rewrite of the Publisher/Subscriber based FileIO onto a GraphStage with AsyncFileChannel - callback handling is so much simpler now that we have `GraphStage` (we did not have it before).

In summary:
- it doesn't, I don't think you'll notice though - dedicated pools for the blocking ops now hide most of the difference
- it could be rewritten, we currently do not have the capacity to hack on it, if you'd like to help out that'd be awesome :-)

**Comment by julianvmodesto** [comment]
Sorry I deleted my comment ("Does the File IO implementation use NIO's AsynchronousFileChannel?") – deleted because I saw #19723 and #17269.

Appreciate the reply! I'm entirely new to Akka hehe. GraphStage sounds interesting, I'll take a look at it.
